src/main/java/com/brandon3055/draconicevolution/client/keybinding/KeyInputHandler.java

```diff
@@ -91,29 +91,15 @@ else if (KeyBindings.toolProfileChange.isPressed() && player != null && HandHelp
 //			}
         }
 
-        int change = Mouse.getEventDWheel();
+        int change = event.getDwheel();
         if (change == 0 || !player.isSneaking()) return;
 
-        if (change > 0) {
-            ItemStack item = player.inventory.getStackInSlot(previouseSlot(1, player.inventory.currentItem));
-            if (item != null && item.getItem() == DEFeatures.dislocatorAdvanced) {
-                player.inventory.currentItem = previouseSlot(1, player.inventory.currentItem);
-                DraconicEvolution.network.sendToServer(new PacketDislocator(PacketDislocator.SCROLL, -1, false));
-            }
-        }
-        else if (change < 0) {
-            ItemStack item = player.inventory.getStackInSlot(previouseSlot(-1, player.inventory.currentItem));
-            if (item != null && item.getItem() == DEFeatures.dislocatorAdvanced) {
-                player.inventory.currentItem = previouseSlot(-1, player.inventory.currentItem);
-                DraconicEvolution.network.sendToServer(new PacketDislocator(PacketDislocator.SCROLL, 1, false));
-            }
+        ItemStack item = player.inventory.getStackInSlot(player.inventory.currentItem);
+        if (item.getItem() == DEFeatures.dislocatorAdvanced) {
+            if (event.isCancelable())
```

Comment:
This is redundant, You should know what the event is and if its cancelable when subscribing to it.

Reply:
I have to agree with covers on this one. You do not need to check isCancelable when you know the event is cancelable. Please remove that call then if your changes work and fix the buginess with switching locations i will pull it.